modules/videoio/src/cap_ffmpeg_impl.hpp

```diff
@@ -1760,6 +1769,8 @@ double CvCapture_FFMPEG::getProperty( int property_id ) const
     case CAP_PROP_STREAM_OPEN_TIME_USEC:
         //ic->start_time_realtime is in microseconds
         return ((double)ic->start_time_realtime);
+    case CAP_PROP_N_THREADS:
+        return static_cast<double>(nThreads);
```

Comment:
Could we retrive `enc->thread_count` value here?

Reply:
> BTW, FFmpeg emits warning about threads on i7-12700K (20 threads)

I don't see a warning on a i7-12700H from `cv::VideoCapture` with FFMpeg backend do you mean the ffmpeg application?


> Could we retrive `enc->thread_count` value here?

Yes, that's a much better idea as the requested value can be overidden in calls to `avcodec_open2`.

Reply:
Try `OPENCV_FFMPEG_DEBUG=1` env variable.

Reply:
Cool cheers, 
> [OPENCV:FFMPEG:24] Application has requested 20 threads. Using a thread count greater than 16 is not recommended.

The comment
https://github.com/FFmpeg/FFmpeg/blob/70f06dd63c9967881c399662646cc593cc5920fb/libavcodec/pthread_internal.h#L24
implies this is h264 only but I get the warning for all file types.  Do you think slice threading is refering to time slice threading and performance or actual errors in the encode/decode?

Do you think this should be hardcoded (`const int nCpus = min(16,get_number_of_cpus());`), it may change and its possible that using more than the default number of threads may have a performance advantage at least for encoding.
https://trac.ffmpeg.org/ticket/8694